src/network/network_gui.cpp

```diff
@@ -233,6 +234,8 @@ class NetworkGameWindow : public Window {
 	int blot_offset; ///< Left offset for green/yellow/red compatibility icon.
 	int flag_offset; ///< Left offset for language flag icon.
 
+	byte widget_id; ///< The widget that has the pop-up input menu.
```

Comment:
This could be named better. Suggesting `query_owner` although that's a bit long.
Really maybe `ShowQueryString` should be changed to take a std::function instead of an object.

Reply:
I just copied what the other windows already were doing :D But I do agree ;)